cmd/maya-exporter/app/collector/metrics.go

```diff
@@ -127,112 +137,211 @@ type MetricsDiff struct {
 // CstorStatsExporter.
-func MetricsInitializer() *Metrics {
+func MetricsInitializer(casType string) *Metrics {
 	return &Metrics{
-		actualUsed: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "actual_used",
-			Help:      "Actual volume size used",
-		}),
-
-		logicalSize: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "logical_size",
-			Help:      "Logical size of volume",
-		}),
-
-		sectorSize: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "sector_size",
-			Help:      "sector size of volume",
-		}),
-
-		readIOPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "read_iops",
-			Help:      "Read Input/Outputs on Volume",
-		}),
-
-		readTimePS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "read_time_per_second",
-			Help:      "Read time on volume per second",
-		}),
-
-		readBlockCountPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "read_block_count_per_second",
-			Help:      "Read Block count of volume per second",
-		}),
-
-		writeIOPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "write_iops",
-			Help:      "Write Input/Outputs on Volume per second",
-		}),
-
-		writeTimePS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "write_time_per_second",
-			Help:      "Write time on volume per second",
-		}),
-
-		writeBlockCountPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "write_block_count_per_second",
-			Help:      "Write Block count of volume per second",
-		}),
-
-		readLatency: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "read_latency",
-			Help:      "Read Latency count of volume",
-		}),
-
-		writeLatency: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "write_latency",
-			Help:      "Write Latency count of volume",
-		}),
-
-		avgReadBlockCountPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "avg_read_block_count_per_second",
-			Help:      "Average Read Block count of volume per second",
-		}),
-
-		avgWriteBlockCountPS: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "avg_write_block_count_per_second",
-			Help:      "Average Write Block count of volume per second",
-		}),
-
-		sizeOfVolume: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "size_of_volume",
-			Help:      "Size of the volume requested",
-		}),
-
-		volumeUpTime: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: "OpenEBS",
-			Name:      "volume_uptime",
-			Help:      "Time since volume has registered",
-		},
+		totalReadBytes: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: "OpenEBS",
```

Comment:
Namespace should be set to the actual namespace where the volume controller and exporter are running. This can be addressed in a separate PR.

Reply:
@avishnu There are two approaches to get the namespace by the application
1. [Use downward API's](https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/downward-api.md) (simple and easiest ones)
2. Make call to kube-apiserver to get the namespace by use of kubernetes client.